### PR TITLE
Feature/substudy patient home updates

### PR DESCRIPTION
**Pull request by achen2401**
Address:
https://jira.movember.com/browse/TN-2689
https://jira.movember.com/browse/TN-2691
Additional of sub-study tiles on the patient home page and sub-study post questionnaire thank you modal
Changes include:

- addition of sub-study tile(s) IF subject is determined to be part of the sub-study
- addition of sub-study due tile IF sub-study questionnaire is due, in-progress or overdue.  The tile **will be greyed out** IF main-study questionnaire is not completed yet.
- addition of sub-study completed tile IF sub-study questionnaire has been completed.
- addition of thank you modal when sub-study is completed
- tile title change for subjects enrolled in sub-study

Noting that modification of trigger filtered content in thank you modal requires API from here:  https://github.com/uwcirg/truenth-portal/pull/3814 but it should not be a blocker for this one to be merged.

### CODE REVIEW

**Comment by achen2401** [comment]
@pbugni Your preliminary review on this is much appreciated.

File: portal/eproms/views.py [hunk 1]
```diff
@@ -137,7 +156,13 @@ def assessment_engine_view(user):
...
         expired_date=expired_date,
         assessment_is_due=assessment_is_due,
-        comp_date=comp_date
+        comp_date=comp_date,
+        enrolled_in_substudy=enrolled_in_substudy,
+        substudy_assessment_status=substudy_assessment_status,
+        substudy_assessment_is_due=substudy_assessment_is_due,
+        substudy_expired_date=substudy_expired_date,
+        substudy_comp_date=substudy_comp_date
+
```
Comment (pbugni) [inline portal/eproms/views.py:165]: remove this blank line, please.

**Review by pbugni — APPROVED** [review]
Looks like you're on the right track to me!
Keep in mind, we will have users enrolled in the substudy without a well defined due date if we're waiting for them to complete the respective global study first.

File: portal/static/js/src/empro.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import tnthAjax from "./modules/TnthAjax.js";
+
+var emproObj = function() {
+    this.domains = [];
+    this.hasHardTrigger = false;
+    this.hasSoftTrigger = false;
+    this.userId = 0;
+};
+emproObj.prototype.populateDomainDisplay = function() {
+    this.domains.forEach(domain => {
```
Comment (pbugni) [inline portal/static/js/src/empro.js:11]: given the fact that multiple domains map to a single display - do you need to confirm you haven't already added, so we don't say get `mood_changes` twice?

File: portal/static/js/src/modules/TnthAjax.js [hunk 3]
```diff
@@ -314,6 +314,24 @@ export default { /*global $ */
             }
         });
     },
+    getSubStudyTriggers: function(userId, params, callback) {
+        callback = callback || function() {};
+        if (!userId) {
+            callback({error: i18next.t("User id is required.")});
+            return false;
+        }
+        this.sendRequest(`/api/user/${userId}/triggers`, "GET", userId, params, function(data) {
```
Comment (pbugni) [inline portal/static/js/src/modules/TnthAjax.js:323]: Great!  But I believe this is an implicit dependency on #3814?

**Review by pbugni — APPROVED** [review]

File: portal/static/js/src/empro.js (hunk 2 above)
Comment (achen2401) [reply portal/static/js/src/empro.js:11]: yeah, I removed duplication(s) here:  https://github.com/uwcirg/truenth-portal/blob/cd5dff64e6d014719a7299e6a446dce06962c16c/portal/static/js/src/empro.js#L32

File: portal/static/js/src/modules/TnthAjax.js (hunk 3 above)
Comment (achen2401) [reply portal/static/js/src/modules/TnthAjax.js:323]: YES, but that is for the thank you modal, which needs more work (https://jira.movember.com/browse/TN-2644) UI wise. But that dependency shouldn't prevent the sub-study tiles from being rendered.

File: portal/eproms/templates/eproms/assessment_engine/ae_base.html [hunk 4]
```diff
@@ -3,4 +3,4 @@
 </section>
 <section class="portal-main portal-flex-container">
     {%- block body -%}{%- endblock -%}
-</section>
+</section>
```
[the remark this answers is not in the document]
Comment (achen2401) [reply portal/eproms/templates/eproms/assessment_engine/ae_base.html]: thank you, added

File: portal/eproms/templates/eproms/assessment_engine/ae_macros.html [hunk 5]
```diff
@@ -52,3 +61,143 @@ <h4 class="portal-description-title">
...
+                        substudy_expired_date=substudy_expired_date,
+                        registry=_(assessment_status.assigning_authority))}}</div>
+                {%- else -%}
+                    <div class="text-content">{{_("Complete your %(substudy_title)s questionnaire here. Please complete your %(registry)s Questionnaire before being able to complete the sub-study questionnaire.", 
+                        substudy_title=substudy_title,
+                        substudy_expired_date=substudy_expired_date,
+                        registry=_(assessment_status.assigning_authority))}}</div>
+                {%- endif -%}
+            {%- else -%}
+                <div class="text-content">{{_("Please complete your questionnaire here.  Your responses will create a big impact for future generations of men with Prostate Cancer.")}}</div>,
```
[the remark this answers is not in the document]
Comment (achen2401) [reply portal/eproms/templates/eproms/assessment_engine/ae_macros.html]: thank you for catching that! fixed.

File: portal/eproms/templates/eproms/assessment_engine/ae_macros.html [hunk 6]
```diff
@@ -52,3 +61,143 @@ <h4 class="portal-description-title">
...
+                    <!-- hard trigger items are hidden by default -->
+                    <div class="item hard-trigger-item">
+                        <!-- TODO dynamic populate domain topics from call to API !?! -->
+                        <p>{{_("You reported having some particular challenges with: ")}}</p>
+                        <ul id="hardTriggerDisplayList">
+                            <!-- TODO this is the subject's domain topic(s)-->
+                            <!--<li>{{_("insomnia")}}</li>-->
+                        </ul>
+                        <br/>
+                        <p><b>{{_("To help address these, we’ve informed your care team and they’ll be in contact with you soon.")}}</b></p>
```
[the remark this answers is not in the document]
Comment (achen2401) [reply portal/eproms/templates/eproms/assessment_engine/ae_macros.html]: thank you for catching that, fixed

File: portal/eproms/templates/eproms/assessment_engine/ae_macros.html [hunk 7]
```diff
@@ -52,3 +61,143 @@ <h4 class="portal-description-title">
...
+                        <br/>
+                        {% if user and user.organizations %}
+                            <p>{{_("In the meantime, if you have any questions or need assistance, please contact your team directly at")}}</p>
+                            <p><b>{{user.organizations[0].name}}</b></p>
+                        {% endif %}
+                    </div>
+                    <!-- TODO show this section only if there is hard trigger -->
+                    <!-- TODO show this section only if there is soft trigger -->
+                    <div class="item hard-trigger-item">
+                        <p>{{_("Based on your questionnaire responses, we’ve also gathered information and resources you may find helpful.")}}</p>
```
[the remark this answers is not in the document]
Comment (achen2401) [reply portal/eproms/templates/eproms/assessment_engine/ae_macros.html]: thank you, fixed

**Pull request merged**